src/App.tsx

```diff
@@ -0,0 +1,221 @@
+import { DetailsPanel } from "./weatherDetails/DetailsPanel";
+import { CurrentWeatherPanel } from "./currentWeather/CurrentWeatherPanel";
+import { createRoot } from "react-dom/client";
+import { useEffect, useState } from "react";
+import { DotSpinner } from "@uiball/loaders";
+
+const baseUrl = "http://api.openweathermap.org";
+const apiKey = "52cd117b0f18fb74a6f94c5c52c15753";
+
+type GcsValues = {
+  lat: string;
+  lon: string;
+};
+
+export type CurrentWeather = {
+  main: { temp: number; humidity: number; pressure: number };
+  weather: [{ description: string; icon: string }];
+  dt: number;
+  name: string;
+  sys: { country: string };
+  wind: { speed: number; deg: number };
+  visibility: number;
+};
+
+export type ForecastWeather = {
+  list: [
+    {
+      dt: number;
+      weather: [{ icon: string; main: string }];
+      main: { temp_max: number; temp_min: number };
+    }
+  ];
+};
+
+type OnError = (error: Error | null) => void;
+
+const initialGcsValues: GcsValues = {
+  lat: "50.45",
+  lon: "30.52",
+};
+
+async function fetchCurrentWeather<T extends Record<string, unknown>>(
+  gcsValues: GcsValues,
+  units: string,
+  onError: OnError
+) {
+  onError(null);
```

Comment:
This is incorrect. `onError` callback should be invoked only when error is got, but not at the start of fetching

Reply:
Then how should i reset fetchError value?